# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../binutils/i686-nopl.patch
# Copyright (C) 2011 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Despite months of prominent people arguing, this again goes thru half fixed,
sigh!

By default GCC (and glibc Makefiles) do not pass -march= down to the
assembler, and -mtune=i686 only let the UNKNOWN march path choose the
wrong nopl from the i686 marked PENTIUM_PRO entry, ... I choose to just
touch the "march" not selected case, as changing the i686 entry to
GENERIC32, or so, may introduce other regressions related to CMOV, et al.

  - Rene Rebe <rene@exactcode.de>

PS: could have just wiped the code, but the #if 0 keeps the patch compact.

--- binutils-2.20.51.0.11/gas/config/tc-i386.c.orig	2011-01-13 12:52:38.724377980 +0000
+++ binutils-2.20.51.0.11/gas/config/tc-i386.c	2011-01-13 12:50:57.863374773 +0000
@@ -1027,15 +1027,18 @@
       if (fragP->tc_frag_data.isa == PROCESSOR_UNKNOWN)
 	{
 	  /* PROCESSOR_UNKNOWN means that all ISAs may be used.  */
+#if 0
 	  switch (cpu_arch_tune)
 	    {
 	    case PROCESSOR_UNKNOWN:
+#endif
 	      /* We use cpu_arch_isa_flags to check if we SHOULD
 		 optimize with nops.  */
 	      if (fragP->tc_frag_data.isa_flags.bitfield.cpunop)
 		patt = alt_long_patt;
 	      else
 		patt = f32_patt;
+#if 0
 	      break;
 	    case PROCESSOR_PENTIUMPRO:
 	    case PROCESSOR_PENTIUM4:
@@ -1061,6 +1064,7 @@
 	      patt = f32_patt;
 	      break;
 	    }
+#endif
 	}
       else
 	{
